src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/BrotliPrimitives.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#if BIT64
+    using nuint = System.UInt64;
+#else
+    using nuint = System.UInt32;
+#endif 
+
+namespace System.IO.Compression
+{
+    public struct BrotliPrimitives
+    {
+        const int defQuality = 11;
+        const int defLgWin = 24;
+
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten)
```

Comment:
I agree these should be int at the public layer. The reasoning is to match similar APIs (base64, encoders, etc) but also because Span only supports int for indexer and thus only allows access to int.MaxValue units in the Span.

I think we should merge this PR. We have another change coming that is going to change the signatures of the public surface to follow a new convention similar to our other Span APIs.